tests/unit/type/list-3.tcl

```diff
@@ -26,32 +26,52 @@ start_server {
         r ping ; # It's enough if the server is still alive
     } {PONG}
 
-    test {Stress tester for #3343-alike bugs} {
+   test { check compression with recompress } {
         r del key
-        for {set j 0} {$j < 10000} {incr j} {
-            set op [randomInt 6]
-            set small_signed_count [expr 5-[randomInt 10]]
-            if {[randomInt 2] == 0} {
-                set ele [randomInt 1000]
-            } else {
-                set ele [string repeat x [randomInt 10000]][randomInt 1000]
-            }
-            switch $op {
-                0 {r lpush key $ele}
-                1 {r rpush key $ele}
-                2 {r lpop key}
-                3 {r rpop key}
-                4 {
-                    catch {r lset key $small_signed_count $ele}
+        config_set list-compress-depth 1
+        config_set list-max-ziplist-size 16
+        r rpush key a
+        r rpush key [string repeat b 50000]
+        r rpush key c
+        r lset key 1 d
+        r rpop key
+        r rpush key [string repeat e 5000]
+        r linsert key before f 1
+        r rpush key g
+   }
+
+    set cycles 1000
+    if {$::accurate} { set cycles 10000 }
+    foreach comp {2 1 0} {
+        config_set list-compress-depth $comp
+        test {Stress tester for #3343-alike bugs comp: $comp} {
+            r del key
+                for {set j 0} {$j < $cycles} {incr j} {
+                set op [randomInt 7]
+                set small_signed_count [expr 5-[randomInt 10]]
+                if {[randomInt 2] == 0} {
+                    set ele [randomInt 1000]
+                } else {
+                    set ele [string repeat x [randomInt 10000]][randomInt 1000]
                 }
-                5 {
-                    set otherele [randomInt 1000]
-                    if {[randomInt 2] == 0} {
-                        set where before
-                    } else {
-                        set where after
+                switch $op {
+                    0 {r lpush key $ele}
+                    1 {r rpush key $ele}
+                    2 {r lpop key}
+                    3 {r rpop key}
+                    4 {
+                        catch {r lset key $small_signed_count $ele}
+                    }
+                    5 {
+                        set otherele [randomInt 1000]
+                        if {[randomInt 2] == 0} {
+                            set where before
+                        } else {
+                            set where after
+                        }
+                        r linsert key $where $otherele $ele
                     }
-                    r linsert key $where $otherele $ele
+                    6 {r lrem key 1 $ele}
```

Comment:
i don't think we wanna use `$ele` here, it's a random value, we need a random "real" element.
maybe get a random element using LLEN, randomInt, and LINDEX?